mbuild/compound.py

```diff
@@ -324,6 +338,52 @@ def particles_by_element(self, element):
             if particle.element == element:
                 yield particle
 
+    @property
+    def mass(self):
+        """Return the total mass of a compound.
+
+        If the compound contains children compouds, the total mass of all
+        children compounds is returned.
+        If the compound contains element information (Compound.element) then
+        the mass is inferred from the elemental mass.
+        If Compound.mass has been set explicitly, then it will override the
+        mass inferred from Compound.element.
+        If neither of a Compound's element or mass attributes have been set,
+        then a mass of zero is returned.
+        """
+        if self._mass is not None:
+            return self._mass
+        else:
+            return sum([self._particle_mass(p) for p in self.particles()])
+
+    def _particle_mass(self, particle):
+        if particle._mass:
+            return particle._mass
+        else:
+            if particle.element:
+                return particle.element.mass
+            else:
+                return 0
+
+    @mass.setter
+    def mass(self, value):
+        if self.children:
+            raise MBuildError(
+                "Cannot set the mass of a Compound containing "
+                "children compounds"
+            )
```

Comment:
I think this condition check should be more specific, since `mb.Port` can be added direct to a particle and stored under that particle's children list.

Reply:
Good catch, I updated the  mass setter to use the same method as the charge setter for checking for children compounds. Let me know if you think that solves this.